internal/server/iigo.go

```diff
@@ -22,10 +22,13 @@ func (s *SOMASServer) runIIGOEndOfTurn() error {
 	clientMap := getNonDeadClients(s.gameState.ClientInfos, s.clientMap)
 	for clientID, v := range clientMap {
 		tax := v.GetTaxContribution()
-		s.gameState.CommonPool += tax
-		newGameState := s.gameState.GetClientGameStateCopy(clientID)
-		newGameState.ClientInfo.Resources -= tax
-		v.GameStateUpdate(newGameState)
+
+		err := s.takeResources(clientID, tax, "tax")
```

Comment:
Potentially need to update the client through calling a function similar to ForageUpdate(), maybe called TaxUpdate(). This would tell the client how much tax was paid, and prompt them to update their gamestate

Reply:
Added